airbyte-cdk/python/airbyte_cdk/sources/declarative/incremental/datetime_based_cursor.py

```diff
@@ -100,6 +102,8 @@ def update_cursor(self, stream_slice: StreamSlice, last_record: Optional[Record]
         stream_slice_value = stream_slice.get(self.cursor_field.eval(self.config))
         stream_slice_value_end = stream_slice.get(self.partition_field_end.eval(self.config))
         last_record_value = last_record.get(self.cursor_field.eval(self.config)) if last_record else None
+        if last_record_value and self.cursor_datetime_format:
+            last_record_value = self._format_datetime(self._parser.parse(last_record_value, self.cursor_datetime_format, self._timezone))
```

Comment:
The `_format_datetime()` call will ultimately yield a string. We touched on a bit during slack, but we should just stick to comparing datetime objects against each other instead of strings. 

In addition to this field, we'll also need to make sure in https://github.com/airbytehq/airbyte/pull/21956/files#diff-51246ab76c66444ad999d4d0e0b1590c6b1814f3d1b27b9d9441b999583a6e0dR90-R91
ee convert the `self._cursor` datetime object back into a string when we emit the state message so we'll need to call the `_format_datetime()` in that method.

Reply:
yeah I think it is fair to scope this to another ticket. I'll write up a ticket for a follow up